src/code/z_view.c

```diff
@@ -51,80 +51,88 @@ void View_Init(View* view, GraphicsContext* gfxCtx) {
     view->up.z = 0.0f;
     view->eye.z = -1.0f;
 
-    if (D_8012ABF0) {
-        if (D_8012ABF0 == 0) {}
+    if (sViewNotInitialized) {
+        if (sViewNotInitialized == 0) {}
         osSyncPrintf("\nview: initialize ---\n");
-        D_8012ABF0 = false;
+        sViewNotInitialized = false;
     }
 
     view->unk_124 = 0;
-    view->flags = 1 | 2 | 4;
+    view->dirtyFlags = VIEW_VIEWING | VIEW_VIEWPORT | VIEW_PROJECTION_PERSPECTIVE;
     View_InitDistortion(view);
 }
 
-void func_800AA358(View* view, Vec3f* eye, Vec3f* lookAt, Vec3f* up) {
+void View_LookAt(View* view, Vec3f* eye, Vec3f* lookAt, Vec3f* up) {
     if (eye->x == lookAt->x && eye->z == lookAt->z) {
```

Comment:
```suggestion
void View_SetViewingEyeAtUp(View* view, Vec3f* eye, Vec3f* at, Vec3f* up) {
    if (eye->x == at->x && eye->z == at->z) {
```
What about something like this?

Reply:
To be honest, I do rather prefer `LookAt` here. It's concise and is something of a standard name for setting a view transform from an eye position, target position and up direction. (c.f. `guLookAt`, `gluLookAt`.)

Reply:
That's fair. `LookAt` feels like the right name to me but it's just preference. Consistency within this repo is probably more important.

Reply:
The big issue I have for a name like `View_SetViewingEyeAtUp` is that the operation being performed feels needlessly lost in a salad of words, that the name choice is driven too much by the arguments being passed into the function instead of being representative of the high level operation that the function performs. I guess I'm also not fully aware of why the concept of lookAt should be totally thrown away in this instance.

Reply:
For some addditional context, every 3D graphics library I've seen that's high-level enough to include a function for setting a view transform from an eye/at/up triple calls it LookAt:

[guLookAt](https://ultra64.ca/files/documentation/online-manuals/man/n64man/gu/guLookAt.html)
[gluLookAt](https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/gluLookAt.xml)
[D3DMatrixLookAtLH/RH](https://docs.microsoft.com/en-us/windows/win32/direct3d9/d3dxmatrixlookatlh)
[glm::lookAt](https://glm.g-truc.net/0.9.9/api/a00668.html#gaa64aa951a0e99136bba9008d2b59c78e)
[bgfx: mtxLookAt](https://github.com/bkaradzic/bx/blob/master/src/math.cpp#L241)
[Unity: LookAt](https://docs.unity3d.com/ScriptReference/Matrix4x4.LookAt.html)
[Unreal: Look At](https://docs.unrealengine.com/4.26/en-US/BlueprintAPI/Utilities/Animation/LookAtFunction/)

Given that this is such a ubiquitous operation in CG, and one with such a standardized name, it would seem like a shame to just name it after its arguments.

Consistency within this repo is important too, though, so there is an argument to be made for using the name used in the decompiled camera code name here and perhaps considering changing everything to LookAt later.

Reply:
If it is the common term used everywhere, then we may need to discuss introducing that into the camera docs. But that will need some coordinating with engineer who has been doing all of the camera stuff. 
So, what are your thoughts @engineer124

Reply:
There is a difference actually, the cited `lookat`s functions are producing matrices, the `View_LookAt` function is only a setter for eye/at/up vectors that are then used in other functions to pass to `guLookAt`

if we still want to go with `LookAt` we may also want to change `View_SetPerspective`/`View_SetOrtho` to match `guPerspective`/`guOrtho` as well. For example in `Title_SetupView` it looks weird:
https://github.com/rogual/oot/blob/46ad1a2cb6bac4c40efa23432a7b03437e328f82/src/overlays/gamestates/ovl_title/z_title.c#L52-L54
```c
    View_SetPerspective(view, 30.0f, 10.0f, 12800.0f);
    View_LookAt(view, &eye, &lookAt, &up);
    View_Apply(view, VIEW_ALL);
```

but I feel like `View_LookAt` not producing a matrix is enough of an argument to not stick to `LookAt` and use `SetEyeAtUp` instead

Reply:
A matrix is one possible representation of a coordinate transform. In my view, `View`'s interface is that it represents a coordinate transform, and the fact that it uses matrices internally (and uses the flag to decide when to update them) is an implementation detail. To "look at" something is to orient oneself. `LookAt` is something that makes sense for any representation of a coordinate transform, not just a matrix. For instance, I would feel that LookAt would be an equally good name even if Euler angles or quaternions were being used.

Reply:
Not all of the cited `LookAt`s work on matrices. They tend to work on whatever representation of a Transform the library in question uses. For GL that's matrices, but for Unity and Unreal that's a `Transform` object which doesn't expose its internal representation.

Reply:
I'm not dead set against LookAt, my main point is instead of (in `Title_SetupView`)
```c
    View_SetPerspective(view, 30.0f, 10.0f, 12800.0f);
    View_LookAt(view, &eye, &lookAt, &up);
    View_Apply(view, VIEW_ALL);
```
we should have either
```c
    View_Perspective(view, 30.0f, 10.0f, 12800.0f);
    View_LookAt(view, &eye, &lookAt, &up);
    View_Apply(view, VIEW_ALL);
```
or
```c
    View_SetPerspective(view, 30.0f, 10.0f, 12800.0f);
    View_SetEyeAtUp(view, &eye, &lookAt, &up);
    View_Apply(view, VIEW_ALL);
```

Reply:
don't understand why `View_SetLookAt` isn't also a valid option; view's eye, at, up properties appear to be used exclusively for calculating the LookAt matrix.

Reply:
`SetLookAt` is fine for me too. Slight preference for just `LookAt` but I don't really care, I really think we're firmly in nitpick territory now.

Reply:
well my only issue is again the `View_SetPerspective`/`View_LookAt` combo looks weird to me but if I'm the only one I'm ready to drop the matter